pypuf/simulation.py

```diff
@@ -106,3 +106,36 @@ def ltf_eval(self, inputs):
                 for l in range(self.k)
             ])
         )
+
+
+class NoisyLTFArray(LTFArray):
+    """
+    Class that simulates k LTFs with n bits and a constant term each 
+    with noise effect and constant bias (=scale) added.
+    """
+    def __init__(self, weight_array, transform, combiner, scale, noise):
+        """
+        Initializes LTF array with bias and noise effect based on inputs.
+        The noise effect equals a normal distributed random variable
+        with mu=0, sigma= scale of overall weights * noise.
+        The bias is a constant value out of a normal distribution.
+        """
+        super().__init__(weight_array, transform, combiner)
+        self.scale = scale
+        self.bias = random.normal(loc=0, scale=scale, size=self.k)
+        self.sigma_noise = sqrt(scale**2 * self.n) * noise
+
+    def ltf_eval(self, inputs):
+        """
+        :return: array
+        """
+        noise = random.normal(loc=0, scale=self.sigma_noise, size=self.k)
```

Comment:
We need simulations to be pseudo-random, yet reproducible. Hence we need to provide the user with a way to setup the PRNG with a custom seed as done in, for instance, [here](https://github.com/nils-wisiol/pypuf/blob/0c91ff04090541dedee57923262e1c90e61beacc/sim_learn.py#L35). Keep in mind that several parts of pypuf share the same PRNG and we need to avoid interference of the custom seeding with other code that seeds the PRNG; an exclusive PRNG-instance for the simulation could be appropriate.